atproto/firehose/client.py

```diff
@@ -118,55 +123,23 @@ def _get_reconnection_delay(self) -> int:
 
         return min(base_sec, self._max_reconnect_delay_sec) + rand_sec
 
-    def _process_raw_frame(self, data: bytes) -> None:
-        frame = Frame.from_bytes(data)
-        if isinstance(frame, ErrorFrame):
-            raise FirehoseError(XrpcError(frame.body.error, frame.body.message))
-        if isinstance(frame, MessageFrame):
-            self._process_message_frame(frame)
-        else:
-            raise FirehoseDecodingError('Unknown frame type')
-
-    def start(
-        self,
-        on_message_callback: t.Union[OnMessageCallback, AsyncOnMessageCallback],
-        on_callback_error_callback: t.Optional[OnCallbackErrorCallback] = None,
-    ) -> None:
-        """Subscribe to Firehose and start client.
-
-        Args:
-            on_message_callback: Callback that will be called on the new Firehose message.
-            on_callback_error_callback: Callback that will be called if the `on_message_callback` raised an exception.
-
-        Returns:
-            :obj:`None`
-        """
-        self._on_message_callback = on_message_callback
-        self._on_callback_error_callback = on_callback_error_callback
-
-    def stop(self):
-        """Unsubscribe and stop the Firehose client.
-
-        Returns:
-            :obj:`None`
-        """
-        raise NotImplementedError
-
-    def _process_message_frame(self, frame: 'MessageFrame') -> None:
-        raise NotImplementedError
-
 
 class _WebsocketClient(_WebsocketClientBase):
     def __init__(
         self, method: str, base_uri: t.Optional[str] = None, params: t.Optional[t.Dict[str, t.Any]] = None
     ) -> None:
         super().__init__(method, base_uri, params)
 
+        # TODO: Not sure if this should be a Lock or not, the async is using an Event now
         self._stop_lock = threading.Lock()
```

Comment:
let's rework to `threading.Event`?

Reply:
Not entirely confident doing that since I'm not super familiar with the details of threading in python, and I think that's getting towards scope-creep for this PR.

Reply:
> I think that's getting towards scope-creep for this PR.

maybe but i don't want to bring features/improvements only in async version. if there is an improvement it must be done for both versions :( 

let's remove wait_for for async version from this PR